java-frontend/src/main/java/org/sonar/java/model/JMethodSymbol.java

```diff
@@ -96,45 +99,57 @@ public TypeSymbol returnType() {
   @Nullable
   @Override
   public MethodSymbol overriddenSymbol() {
-    if (overriddenSymbol == Symbols.unknownMethodSymbol) {
-      overriddenSymbol = convertOverriddenSymbol();
+    if (overriddenSymbols == null) {
+      // only compute overridenSymbols once
+      overriddenSymbols();
     }
-    return overriddenSymbol;
+    return firstOverridenSymbol;
   }
 
-  @Nullable
-  private MethodSymbol convertOverriddenSymbol() {
-    IMethodBinding overrides = find(
-      methodBinding()::overrides,
-      methodBinding().getDeclaringClass()
-    );
-    if (overrides == null) {
-      return null;
+  @Override
+  public List<MethodSymbol> overriddenSymbols() {
```

Comment:
In the different rules using this method, we often use the first element of the list. However, this list is internally built thanks to a Set, I'm afraid we could introduce inconsistency due to the fact that the first element is not always the same one.
Same concern for `firstOverridenSymbol`, it will be the same between two calls, but not necessarily between two analyses.

Reply:
The set(s) used to build the list is done through `linkedHashSet`, and preserve the order of insertion.
Also, the list of parents types is always explored in the same order, being the declaration order.

Reply:
Make sense, thanks for the clarification.